chain/reorg.go

```diff
@@ -125,13 +193,65 @@ func (cs *ChainService) reorg(topBlock *types.Block) error {
 //  - tx mapping
 //  - best block
 func (reorg *reorganizer) swapChain() error {
-	cs := reorg.cs
+	cdb := reorg.cs.cdb
 
 	logger.Info().Msg("swap chain to new branch")
 
-	cs.cdb.swapChain(reorg.newBlocks)
+	marker := NewReorgMarker(reorg)
```

Comment:
Is a new marker needed for a recovery case?

In the case of a recovery, the marker seems to be loaded from the chain DB and dumped. In my view, here we check whether reorg has a valid marker in it and (if it has) use it instead of creating a new one.

Reply:
I have refactored reorg marker as you have advised. 
I added read/write api to marker. and I fixed that marker is made in gatherPostFn after gather() during reorg.
Thanks!